extra_data/reader.py

```diff
@@ -199,6 +205,20 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             for file in self.files:
                 file.close()
 
+    @property
+    def selection(self):
+        # This was previously a regular attribute, which code may have relied on.
+        return {src: srcdata.sel_keys for src, srcdata in self._sources_data.items()}
+
+    @property
+    def _source_index(self):
+        warn(
+            "DataCollection._source_index will be removed. "
+            "Contact da-support@xfel.eu if you need to discuss alternatives.",
```

Comment:
If anyone is using this, they can just replace `run._source_index[source]` with `run[source].files`. We could just put that in the warning. I've suggested that people contact us partly so we can get a bit of a sense if anyone is using it, and partly so we can explain about not relying on things that `_start_with_underscore`.